### ISSUE TITLE
Documentation hard to follow, I cannot get the sample working.

### ISSUE THREAD

**Issue by adrichema** [giuleon/O365ModernProvisioning#1]
Let me first of all complement you with building this solution. It looks very promising! 

I am trying to make this sample work on our Azure environment but unfortunately the documentation is quite hard to follow. Especially the steps further down the lane are not described in detail. 

**What did I do?**

- Added a application to Azure AD
- Deployed the bot to a Bot Web App service
- Deployed the Azure function
- Created a Key Vault
- Created the list in SharePoint

**Problems with Veronica Bot**
I am able to get the bot running, but it fails at the askForOwner step. The bot simply hangs and is not listening for any other message. Cleaning the state (in the table storage) allows me to restart the conversation flow. Can you give me any tips to debug or find out what Is causing the error?  

Does the documentation describe the right permissions? I see that 3 permissions on the Microsoft Graph API are selected in the screenshot, but the text only talks about 2. I am happy to help improving the documentation after I got the sample working. What are the exact settings for the Azure AD registration? 

**Problems with Azure Fuction**
The Azure Function is failing as well. How to connect the Azure Key Vault to the Azure Function for Provisioning? I think the application setting SiteUrl is missing from the documentation.

**Comment by giuleon** [comment]
Sorry for the big delay, I appreciate your feedback because is very helpful in order to improve the documentations.
Regarding the Azure Function you have to get the Url from Azure Key Vault just created:

In the Key Vault context click on secrets:
![image](https://user-images.githubusercontent.com/3914090/50327843-fc1e0a00-04f0-11e9-889e-ecfc344b3998.png)

click on the name of your secret and finally you'll see the Url that you have to paste in the Azure Function environment variable:
![image](https://user-images.githubusercontent.com/3914090/50327964-69319f80-04f1-11e9-9586-950cf2da831f.png)

Now go in the application settings of the Azure Function and create the variable "KeyVaultSecret" and paste here the secret Url:
![image](https://user-images.githubusercontent.com/3914090/50328022-a26a0f80-04f1-11e9-9cf0-587159f5ce99.png)

This variable will be read in the Azure Function at the following line:
adminPassword = (await kvClient.GetSecretAsync(Environment.GetEnvironmentVariable("KeyVaultSecret"))).Value;

The MS Graph permission are fine with these two, if you want to debug the Bot you need to install Bot emulator, but before to do that you can try to test the entire process by adding a new item in the list.
